libnvme-rs/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+
+#[allow(clippy::all)]
+mod bindings {
+    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+}
+pub use bindings::*;
+
+pub mod error;
+pub mod nvme_namespaces;
+mod nvme_tree;
+mod nvme_uri;
+
+use error::{IoError, NvmeError};
+use snafu::ResultExt;
+use std::{fs, path::Path, str::FromStr};
+
+pub use nvme_uri::NvmeTarget;
+
+/// Read and parse value from a sysfs file
+pub fn parse_value<T>(dir: &Path, file: &str) -> Result<T, NvmeError>
```

Comment:
We already have a similar function in [sysfs](https://github.com/openebs/mayastor/blob/322aac84f19c0fcb12b422dd35bd5ff4ec8f44ac/sysfs/src/lib.rs#L13) crate. Can it be reused?

Reply:
This function comes from `nvmeadm` and is in fact newer, with revised error results.